exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/ParquetReaderUtility.java

```diff
@@ -181,6 +185,71 @@ else if (parquetTableMetadata instanceof Metadata.ParquetTableMetadata_v2 &&
   }
 
   /**
+   * Checks that the metadata file was created by drill with version less than
+   * the version where was changed the serialization of BINARY values
```

Comment:
Can you explain a bit more about why the serialization of BINARY values are wrong in prior version? Is it only happening in the metadata cache file?